### PR TITLE
Add `multiple-receive` bbe

### PR DESCRIPTION
**Pull request by poorna2152**
## Purpose
$subject

Part of ballerina-platform/ballerina-dev-website#8856

### CODE REVIEW

File: examples/multiple-receive/multiple_receive.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Multiple receive
+
+The multiple receive action allows receiving messages from multiple workers.
```
Comment (nipunayf) [inline examples/multiple-receive/multiple_receive.md:3]: The sentence doesn't effectively differentiate between the use cases of multiple receives and alternate receive. Can we mention how it waits for all the values and constructs the map structure in a short sentence?

File: examples/index.json [hunk 2]
```diff
@@ -1467,6 +1467,13 @@
         "verifyBuild": true,
         "verifyOutput": true,
         "isLearnByExample": true
+      },
+      {
+        "name": "Multiple receive",
```
Comment (lochana-chathura) [inline examples/index.json:1472]: The correct order is https://github.com/ballerina-platform/ballerina-dev-website/issues/8856#issuecomment-2044468508
Need to move this next to the `Inter-worker message passing`

File: examples/multiple-receive/multiple_receive.metatags [hunk 3]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates the multiple receive action.
```
Comment (lochana-chathura) [inline examples/multiple-receive/multiple_receive.metatags:1]:
```suggestion
description: This BBE demonstrates the use of the multiple receive action in inter-worker communication
```

File: examples/multiple-receive/multiple_receive.bal [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    worker w2 {
+        runtime:sleep(2);
+        3 -> w3;
+    }
+
+    worker w3 returns map<int> {
+        // The worker waits until both values are received.
+        map<int> result = <- {w1, w2};
```
Comment (lochana-chathura) [inline examples/multiple-receive/multiple_receive.bal:16]: Shall we use a record here similar to [multiple-wait](https://ballerina.io/learn/by-example/multiple-wait/)? IMO, A record having two fields to hold the received data is more appropriate and makes the usage more clear.

File: examples/multiple-receive/multiple_receive.bal [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    worker w3 returns Result {
+        // The worker waits until both values are received.
+        Result result = <- {w1, w2};
+        return result;
+    }
+
+    Result result = wait w3;
+    io:println(result.w1);
+    io:println(result.w2);
```
Comment (lochana-chathura) [inline examples/multiple-receive/multiple_receive.bal:27]: Let's directly print the record. 
```suggestion
    io:println(result);
```

File: examples/multiple-receive/multiple_receive.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Multiple receive
+
+The multiple receive action enables receiving messages from multiple workers, waiting until a message is available from each worker to construct a record with received values.
```
Comment (lochana-chathura) [inline examples/multiple-receive/multiple_receive.md:3]: HBT?
```suggestion
The multiple receive action can be used to receive values from multiple send actions. It operates by awaiting the receipt of values from all the send actions, subsequently constructing a map containing those values.
```

**Review by lochana-chathura — APPROVED** [review]

**Review by lochana-chathura — DISMISSED** [review]

**Review by lochana-chathura — DISMISSED** [review]

**Review by SasinduDilshara — APPROVED** [review]

File: examples/multiple-receive/multiple_receive.bal [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        runtime:sleep(2);
+        3 -> w3;
+    }
+
+    worker w3 returns json {
+        // The worker waits until both values are received.
+        Result result = <- {a: w1, b: w2};
+        return result.toJson();
+    }
+
```
Comment (MaryamZi) [inline examples/multiple-receive/multiple_receive.bal:24]: Can we redo this similar to the alternate receive BBE?

w1 and w2 do two fetches. Once both complete w3 does another fetch, sets a field to the record and returns the value?

```ballerina
type Result record {|
    T a;
    T b;
    T c?;
|};
```

File: examples/multiple-receive/multiple_receive.bal [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    // The worker waits until both values are received.
+    Result result = <- {a: w1, b: w2};
+    io:println(result);
+}
+
+function fetch(string url) returns string|error {
+    http:Client cl = check new (url);
+    Response {args: {'worker: 'worker}} = check cl->get("");
```
Comment (SasinduDilshara) [inline examples/multiple-receive/multiple_receive.bal:31]:
```suggestion
    Response {args: {'worker}} = check cl->get("");
```

File: examples/multiple-receive/multiple_receive.bal [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/lang.runtime;
+
+type Response record {
+    record {string 'worker;} args;
```
Comment (MaryamZi) [inline examples/multiple-receive/multiple_receive.bal:6]: I would format this on to separate lines.
```suggestion
    record {
        string 'worker;
    } args;
```

File: examples/multiple-receive/multiple_receive.bal [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    // The worker waits until both values are received.
+    Result result = <- {a: w1, b: w2};
+    io:println(result);
+}
+
+function fetch(string url) returns string|error {
+    http:Client cl = check new (url);
+    Response {args: {'worker}} = check cl->get("");
+    return 'worker;
+}
```
Comment (MaryamZi) [inline examples/multiple-receive/multiple_receive.bal:33]:
```suggestion
function fetch(string workerParam) returns string|error {
    http:Client cl = check new ("https://postman-echo.com");
    Response response = check cl->/get('worker = workerParam);
    return response.args.'worker;
}
```

**Review by SasinduDilshara — APPROVED** [review]

File: examples/multiple-receive/multiple_receive.bal [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+function fetch(string workerParam) returns string|error {
+    http:Client cl = check new ("https://postman-echo.com");
+    Response response = check cl->/get('worker = workerParam);
+    return response.args.'worker;
+}
+
+public function main() {
+    worker w1 {
```
Comment (MaryamZi) [inline examples/multiple-receive/multiple_receive.bal:23]: Please add some comments similar to the other PR.

File: examples/multiple-receive/multiple_receive.bal [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+function fetch(string workerParam) returns string|error {
+    http:Client cl = check new ("https://postman-echo.com");
+    Response response = check cl->/get('worker = workerParam);
+    return response.args.'worker;
+}
+
+public function main() {
+    // Workers `w1` and `w2` fetch content using the respective arguments.
```
Comment (MaryamZi) [inline examples/multiple-receive/multiple_receive.bal:23]:
```suggestion
    // Workers `w1` and `w2` call the `fetch` function to retrieve content. The workers 
    // send the result of calling the `fetch` function to the default worker.
```

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**